mlflow_extensions/databricks/deploy/ez_deploy_ray_serve.py

```diff
@@ -0,0 +1,216 @@
+    Function to Format the Parameter Arguements given to worflow to string
+    '''
+    return {
+        "ez_deploy_config": config.serialize_json(),
+        "hf_secret_scope": hf_secret_scope or "",
+        "hf_secret_key": hf_secret_key or "",
+        "min_replica": min_replica,
+        "max_replica": max_replica,
+        "gpu_config": json.dumps(gpu_node),
+        "pip_reqs": " ".join(config.engine_config.default_pip_reqs()),
+    }
+
+
+def update_cloud_specific_driver_node(cloud: Cloud):
+    if cloud == Cloud.GCP:
+        return {
+            "driver_node_type_id": "n2-highmem-4"
+        }
+    if cloud == Cloud.AWS:
+        return {
+            "driver_node_type_id": "i3.xlarge"
+        }
+    if cloud == Cloud.AZURE:
+        return {
+            "driver_node_type_id": "Standard_DS3_v2"
+        }
+    raise ValueError(f"Cloud {cloud} is not supported")
+
+
+def make_create_json(
+    *,
+    job_name: str,
+    minimum_memory_in_gb: int,
+    min_replica: int,
+    max_replica: int,
+    cloud_provider: Cloud,
+    ez_deploy_config: "EzDeployConfig",
+    huggingface_secret_scope: str,
+    huggingface_secret_key: str,
+    task_name: str = EZ_DEPLOY_TASK,
+    runtime: str = DEFAULT_RUNTIME,
+    notebook_path: str = DEFAULT_SERVING_NOTEBOOK,
+    specific_git_ref: str = None,
+):
+    vm = JobsConfig(minimum_memory_in_gb=minimum_memory_in_gb).smallest_gpu(
+        cloud_provider
+    )
+    if max_replica == 1:
+        gpu_node = {
+            "spark_version": runtime,
+            "spark_conf": {
+                "spark.master": "local[*, 4]",
+                "spark.databricks.cluster.profile": "singleNode",
+            },
+            "node_type_id": vm.name,
+            "driver_node_type_id": vm.name,
+            "custom_tags": {"ResourceClass": "SingleNode"},
+            "enable_elastic_disk": True,
+            "data_security_mode": "NONE",
+            "runtime_engine": "STANDARD",
+            "num_workers": 0,
+        }
+    else:
+        gpu_node = {
+            "spark_version": runtime,
+            "node_type_id": vm.name,
+            "driver_node_type_id": vm.name,
+            "enable_elastic_disk": True,
+            "data_security_mode": "NONE",
+            "runtime_engine": "STANDARD"
+        }
+        if min_replica == max_replica:
+            gpu_node['num_workers'] = min_replica
+        else:
+            gpu_node["autoscale"] = {
+                "min_workers": min_replica,
+                "max_workers": max_replica
+            }
+        gpu_node.update(update_cloud_specific_driver_node(cloud_provider))
+    gpu_node.update(make_cloud_specific_attrs(cloud_provider))
+
+    if specific_git_ref:
+        parts = specific_git_ref.split("/", maxsplit=1)
+        git_type = "git_branch"
+        if parts[0].startswith("commit"):
+            git_type = "git_commit"
+        elif parts[0].startswith("tag"):
+            git_type = "git_tag"
+
+        git_source = GitSource.from_dict(
```

Comment:
if you can move this over and maybe have this as a argument with the default being this repo because once merged it will be confusing where to apply bugfixes.

Reply:
Makes sense. Will parameterize this and add a default as the base repo

Reply:
Done and tested